drivers/serial/uart_pl011_ambiq.h

```diff
@@ -37,8 +37,6 @@ static inline int pl011_ambiq_clk_set(const struct device *dev, uint32_t clk)
 		clksel = PL011_CR_AMBIQ_CLKSEL_24MHZ;
 		break;
 	case 48000000:
```

Comment:
This line can also be removed now

Reply:
@tgorochowik Required? Or requested?

In the Ambiq HAL and datasheet, we still show the 48MHz option but say reserved.
I was leaving it there to show that it was a limitation of the Ambiq device and not the uart_pl011 driver.

Reply:
@tgorochowik in an effort to get this moving, I have changed the code as you requested.

Reply:
Thank you for understanding.